bazel_tools/client_server_test/runner/Main.hs

```diff
@@ -42,33 +37,31 @@ readPortFile n file =
       readPortFile (n-1) file
     Just p -> pure p
 
-runCheckedProc :: T.Text -> CreateProcess -> IO () -> IO ()
-runCheckedProc desc proc resume =
-  withCheckedProcessCleanup proc $ \(_ :: Handle) stdoutSrc stderrSrc ->
-  withAsync (runConduit (stderrSrc .| splitOutput .| C.awaitForever printStderr)) $ \_ ->
-  withAsync (runConduit (stdoutSrc .| splitOutput .| C.awaitForever printStdout)) $
-  const resume
-
-  where
-    printStdout line =
-      liftIO $ do
-        T.hPutStrLn stdout $ desc <> ": " <> line
-        hFlush stdout
-    printStderr line =
-      liftIO $ T.hPutStrLn stderr $ desc <> ": " <> line
-    splitOutput = C.T.decode C.T.utf8 .| C.T.lines
-
 main :: IO ()
 main = do
   [clientExe, clientArgs, serverExe, serverArgs] <- getArgs
   tempFile <- emptySystemTempFile "client-server-test-runner-port"
```

Comment:
Can we . just use `withTempFile` from Extra and avoid doing our own cleanup.

Reply:
it's cleaner now, thanks